example/EmbeddedChessPieces/src/ECPMovement/ECPMovement.cpp

```diff
@@ -49,44 +64,87 @@ void ECPMovement::turnRight(
     const bool wasRotationSuccessful = rotateToAngle(goalAngle, initialAngle);
 
     delay(MEASURING_DELAY); // for better measuring results
-    const bool isCurrentlyOnWhite = ecpColorDetection.isWhiteField();
-    if (isCurrentlyOnWhite != hasStartedOnWhite || !wasRotationSuccessful) {
-        displayRotionCorrectionRequest(currentField, intendedDirection);
+    const FieldColor currentColor = ecpColorDetection.getFieldColor();
+    if (currentColor != startColor || !wasRotationSuccessful) {
+        displayRotationCorrectionRequest(currentField, intendedDirection);
     }
 };
 
+void ECPMovement::calibrateFieldColor() {
+    ecpColorDetection.calibrateFieldColor();
+};
+
+void ECPMovement::calibrateIRFieldColor() {
+    ecpColorDetection.calibrateIRFieldColor();
+};
+
+void ECPMovement::setColorDetectionMode(bool useIR) {
+    ecpColorDetection.setColorDetectionMode(useIR);
+};
+
+void ECPMovement::setShouldTurnOnColorCorrectionLight(bool turnOn) {
+    ecpColorDetection.setShouldTurnOnColorCorrectionLight(turnOn);
+};
+
 // -----------------------------------------------------------------------------
 // PRIVATE FUNCTIONS
 // -----------------------------------------------------------------------------
 
-void ECPMovement::moveForward(int timeMovement, int timeBreak) {
+void ECPMovement::moveForward(int timeMovement) {
     dezibot.motion.move(0, movementCalibration);
     delay(timeMovement);
     dezibot.motion.stop();
-    delay(timeBreak);
 };
 
-void ECPMovement::moveToNextField() {
-    const bool hasStartedOnWhite = ecpColorDetection.isWhiteField();
-    bool isCurrentlyOnWhite = hasStartedOnWhite;
+bool ECPMovement::moveToNextField() {
+    FieldColor startColor = ecpColorDetection.getFieldColor();
+    if (startColor == 2) {
+        startColor = ecpColorDetection.getLikelyFieldColor();
+    }
+    
+    const FieldColor wantedColor = startColor == 0 ? 
+        (FieldColor) (startColor + 1) : (FieldColor) (startColor -1);
```

Comment:
```suggestion
        (FieldColor) (startColor + 1) : (FieldColor) (startColor - 1);
```
Add white space here

Reply:
Fixed in [08641e5](https://github.com/nicosrm/24-emb-chess/pull/40/commits/08641e5d12de2461df452def41074b6b870ae014) but changed in [57e7495](https://github.com/nicosrm/24-emb-chess/pull/40/commits/57e74956b0712787797ea2ea4d639005a3940e98)